packages/labextension/src/index.ts

```diff
@@ -353,8 +424,43 @@ const extension: JupyterFrontEndPlugin<void> = {
       }
     });
 
-    palette.addItem({ command: "jupytext_metadata", rank: JUPYTEXT_FORMATS.length + 3, category: "Jupytext" });
-  }
+    palette?.addItem({
+      command: "jupytext_metadata",
+      rank: JUPYTEXT_FORMATS.length + 3,
+      category: "Jupytext",
+    });
+
+    // Define file types
+    app.docRegistry.addFileType({
+      name: "jupytext-md",
+      displayName: trans.__("Markdown File"),
+      // Extension file are transformed to lower case...
+      extensions: [".rmd", ".qmd"],
+      icon: markdownIcon
+    });
+
+    // Duplicate notebook factory to apply it on markdown file
+    //   Mirror: https://github.com/jupyterlab/jupyterlab/blob/8a8c3752564f37493d4eb6b4c59008027fa83880/packages/notebook-extension/src/index.ts#L860
+    const factory = new NotebookWidgetFactory({
+      name: "Jupytext Notebook",
+      // label: trans.__("Jupytext Notebook"), // will be needed in JupyterLab 4
+      fileTypes: ["jupytext-md", "markdown", "python"],
```

Comment:
If I understand correctly this will change the default factory for `.py` and `.md` files and will allow the user to open them as notebooks by default.

Many people (incl. myself) will love this, however I am afraid some others will be surprised by this change of behavior, and some will want to differentiate the `.py` and `.md` extensions. 

Could we make this optional? On a few issues we have chatted about using the _document user settings_ for activating the notebook editor as the default viewer for `.md` files, do you think we could use these settings for governing the default editor and viewer here?

Reply:
The attribute `fileTypes` defines which file types are supported by the factory (the factory will for matching types appear in the _Open with..._ submenu). But the default factory is only set for file types specified by the attribute `defaultFor` (see below).

In this PR, the markdown files (+ Rmd and qmd files) are set to default to the notebook factory. But the python files don't - they will still be opened with the default viewer of JupyterLab (aka the file editor).

Tempering with the user settings could be a solution. We could update the default viewer if the user did not customize it (at least I think I know a way). Would you prefer that logic?

Reply:
Thanks @fcollonval 

> In this PR, the markdown files (+ Rmd and qmd files) are set to default to the notebook factory. But the python files don't - they will still be opened with the default viewer of JupyterLab (aka the file editor).

I think this is a reasonable settings, however I think we would surprise (upset) fewer users if we did not change the default factory for any files, unless explicitly required by the user, don't you think?

> Tempering with the user settings could be a solution

Yes that would be great! Still I wonder if this is really in the scope of Jupytext - of course it is where this functionality has been asked for, but maybe this kind of user settings could/should belong to Jupyter Lab, don't you think so? I had the impression that at least for the default viewer such a setting is supposed to exist (but I am not sure it works)

Reply:
Ok then I'll update the PR to not change default viewer for the files and to explain how to change the default viewer from the settings. This is already available through the settings `Document Manager` -> `defaultViewers`.